discretisedfield/field.py

```diff
@@ -335,8 +336,16 @@ def value(self):
 
     @value.setter
     def value(self, val):
-        self._value = val
-        self.array = _as_array(val, self.mesh, self.dim, dtype=self.dtype)
+        if callable(val):
+            nb_value_func = nb.guvectorize(
+                [(nb.float64[:], nb.float64[:])], "(n)->(n)", nopython=True
```

Comment:
What happens if `nopython=True` is wrong? Is it ignored or does it raise an exception?